### ISSUE TITLE
Issues with "consul resolver" and more than 159 backends

### ISSUE THREAD

**Issue by liorfranko** [haproxy/haproxy#1573]
### Detailed Description of the Problem

We use a consul resolver with the following configurations:
```
resolvers consul
        nameserver consul 127.0.0.1:8600
        accepted_payload_size 16834
        resolve_retries       3
        timeout resolve       1s
        timeout retry         3s
        hold other           30s
        hold refused         30s
        hold nx              30s
        hold timeout         30s
        hold valid           30s
backend rv-bidder
        server-template   prod_us.rv-bidder 1-300 _sonic-rv-bidder._prod_us.service.consul check inter 1s fall 3 weight 100 maxconn 800 slowstart 30s resolvers consul resolve-prefer ipv4
```
We found an issue when there are more than 159 backends in the consul's response.
1. Viewing the `/var/lib/haproxy/stats` It looks like only 159 backends are being resolved.
We checked the consul response and saw that the response contains ~200 servers.
We checked the consul's response size, it's less than the `accepted_payload_size`
We are using tcpdump, netstat, and network metrics on the servers and we see that all the 200 are being part of the load-balancing.
We see that the mapping between srv_name to the srv_addr is changing every 30 seconds.
For example:
```
root@rv-bidder-haproxy-i-****:~# echo "show servers state " | socat stdio unix-connect:/var/lib/haproxy/stats | grep 'prod_us.rv-bidder1 '
4 rv-bidder 1 prod_us.rv-bidder1 10.200.248.214 2 64 1 100 186 15 3 4 6 0 0 0 rv-bidder-i-0da**** 8080 _sonic-rv-bidder._prod_us.service.consul
root@rv-bidder-haproxy-i-****:~# echo "show servers state " | socat stdio unix-connect:/var/lib/haproxy/stats | grep 'prod_us.rv-bidder1 '
4 rv-bidder 1 prod_us.rv-bidder1 10.200.197.253 1 64 1 100 18 15 3 4 6 0 0 0 rv-bidder-i-022**** 8080 _sonic-rv-bidder._prod_us.service.consul
```
2. In the UI, we see that only 159 backends are in status UP, while all others are in status DOWN with the error "SOCKERR in 0ms"
3. Every time the mapping between srv_name to srv_addr changes, the slow-start is activated.

It looks like somewhere there is a limit of 159 servers in the consul resolver.

### Expected Behavior

Having more than 159 servers in `/var/lib/haproxy/stats` and in the UI.

### Steps to Reproduce the Behavior

1. Configure haproxy to use consul resolve.
2. Add more than 159 servers in consul under the same SRV.

### What is your configuration?

```haproxy
global
        daemon
        user            haproxy
        group           haproxy
        hard-stop-after 20s
        log             /dev/log len 4096 local2 emerg
        stats socket    /var/lib/haproxy/stats user haproxy group haproxy mode 0770 level admin expose-fd listeners
        chroot          /var/lib/haproxy
        pidfile         /var/run/haproxy.pid
        maxconn         200000

resolvers consul
        nameserver consul 127.0.0.1:8600
        accepted_payload_size 16834
        resolve_retries       3
        timeout resolve       1s
        timeout retry         3s
        hold other           30s
        hold refused         30s
        hold nx              30s
        hold timeout         30s
        hold valid           30s


defaults
        log             global
        option          dontlognull
        no option       log-separate-errors
        no option       dontlog-normal
        errorfile 400   /etc/haproxy/errors/400.http
        errorfile 403   /etc/haproxy/errors/403.http
        errorfile 408   /etc/haproxy/errors/408.http
        errorfile 500   /etc/haproxy/errors/500.http
        errorfile 502   /etc/haproxy/errors/502.http
        errorfile 503   /etc/haproxy/errors/503.http
        errorfile 504   /etc/haproxy/errors/504.http

frontend rv-bidder
        bind            0.0.0.0:80
        mode            http
        option          httplog
        option          splice-auto
        maxconn         20000
        use_backend rv-bidder-qa if rv-bidder-admin-uri

backend rv-bidder
        mode            http
        option          httpchk GET /rv-bidder/healthcheck?slim=true HTTP/1.1\r\nHost:\ www
        http-check      expect status 200
        option          prefer-last-server
        http-reuse      aggressive
        option          redispatch
        retries         0
        timeout queue   500ms
        timeout connect 500ms
        timeout server  2s
        timeout check   1s
        balance         leastconn
        server-template   prod_us.rv-bidder 1-300 _sonic-rv-bidder._prod_us.service.consul check inter 1s fall 3 weight 100 maxconn 800 slowstart 30s resolvers consul resolve-prefer ipv4
```

### Output of `haproxy -vv`

```plain
HA-Proxy version 1.8.16-5c3f237 2018/12/21
Copyright 2000-2018 Willy Tarreau <willy@haproxy.org>

Build options :
  TARGET  = linux2628
  CPU     = generic
  CC      = gcc
  CFLAGS  = -O2 -g -fno-strict-aliasing -Wdeclaration-after-statement -fwrapv -Wno-unused-label
  OPTIONS = USE_GETADDRINFO=1 USE_ZLIB=1 USE_REGPARM=1 USE_OPENSSL=1 USE_PCRE=1 USE_PCRE_JIT=1 USE_NS=1

Default settings :
  maxconn = 2000, bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with OpenSSL version : OpenSSL 1.0.1f 6 Jan 2014
Running on OpenSSL version : OpenSSL 1.0.2n  7 Dec 2017 (VERSIONS DIFFER!)
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : SSLv3 TLSv1.0 TLSv1.1 TLSv1.2
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Encrypted password support via crypt(3): yes
Built with multi-threading support.
Built with PCRE version : 8.31 2012-07-06
Running on PCRE version : 8.39 2016-06-14
PCRE library supports JIT : yes
Built with zlib version : 1.2.8
Running on zlib version : 1.2.11
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with network namespace support.

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available filters :
	[SPOE] spoe
	[COMP] compression
	[TRACE] trace
```

**Comment by git001** [comment]
Your version is quite outdated https://www.haproxy.org/bugs/bugs-1.8.16.html can you try to update the haproxy with the latest 2.5 and see if you still have the issue.

**Comment by TimWolla** [comment]
> can you try to update the haproxy with the latest 2.5

2.5 is not a LTS version. 

@liorfranko Please upgrade to 1.8.30 at the very least. That's the latest version in the 1.8.x branch. But keep in mind that 1.8.x only receives critical fixes and only until the end of the year.

I recommend upgrading to 2.4 (2.4.14 at time of commenting) and then making sure to keep the version upgraded to the latest patch version of 2.4.x, until you are ready to upgrade to the next LTS (2.6).

**Comment by liorfranko** [comment]
With version 2.2.3-0e58a34 I'm having the same issue.
I'll try 2.4.14 and update.

**Comment by TimWolla** [comment]
> With version 2.2.3-0e58a34 I'm having the same issue.

To be clear: 2.2.3 is not the newest 2.2.x (2.2.20 is). Not sure where you got that version from, but it's important to use the latest patch version.

2.4.14 is fine.

**Comment by liorfranko** [comment]
Hi
I had the same behavior with version 2.4.14.
But now using the logs I found that we hit the maximum accepted_payload_size.
We used our own 1.8 haproxy with the following commit:
https://github.com/haproxy/haproxy/commit/9d8dbbc56b73a4473b4e312317692a3d13d1da61

Once increased `DNS_MAX_UDP_MESSAGE` to 65535 the issue is solved.
I'm working on upgrading to 2.4.14 which has the DNS_MAX_UDP_MESSAGE set to 65535 by default.

Thank you for your help.
